File: Challenges/Coding World Cup/2023/Day 22 - ENG vs SL - All Caps.cpp
long long underwaterValves(int n, vector<int> &h)
{
    // Write your code here.
    unordered_map<int, int> map;
    for (int i = 0; i < n; i++)
        map[h[i]] = i;

    sort(h.begin(), h.end());

    long long time = 0;
    int prev = 0;
    for (int i = 0; i < n; i++)
    {
        time += abs(prev - map[h[i]]);
        prev = map[h[i]];
    }

    return time;
}

File: Challenges/Coding World Cup/2023/Day 22 - ENG vs SL - All Caps.java
import java.util.*;

public class Solution  {
    public static long underWaterValves(int n, int []h) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < n; i++)
            map.put(h[i], i);

        Arrays.sort(h);

        long time = 0;
        int prev = 0;
        for (int i = 0; i < n; i++) {
            time += Math.abs(prev - map.get(h[i]));
            prev = map.get(h[i]);
        }

        return time;
    }
}